### PR TITLE
1 stylistic changes

### LINKED ISSUE #1: Stylistic changes to the repo
### Protobuf installation steps missing:
**Arch:** `yay protobuf-c`
**Windows:**
Download for the windows protocol buffers should be available here: https://github.com/protocolbuffers/protobuf/releases/tag/v24.3
Afterwards add the bin folder with the precompiled protoc to your system's enviroment variabels in path, it should compile now after restarting the terminal

## Comments
There are comments in the top of both `src/DataReader/parse_edge.rs` and `src/DataReader/parse_queries.rs` that should be only two '/' instead of three. Otherwise you get information about the entire file on a single function in said file.

There are a few functions inside `src/DataReader/json_reader.rs` that document functions but only has two '/' so the comments dont show up in intellisense.

`src/ModelObjects/state.rs` first comment contains typos, should be 
```rs
/// State is a struct used for initial verification of consistency, and determinism as a state that also hols a dbm
/// This is done as the type used in refinement state pair assumes two sides of an operation
/// this should probably be refactored as it causes unnecessary confusion
```

## Warnings
remove #![allow(non_snake_case)] from [main.rs](https://github.com/ECDAR-AAU-SW-P5/Reveaal/blob/4b41c9daf115183ac615227e3589d00bd99f07f0/src/main.rs#L1) and [lib.rs](https://github.com/ECDAR-AAU-SW-P5/Reveaal/blob/4b41c9daf115183ac615227e3589d00bd99f07f0/src/lib.rs#L1)

## Rename
`src/EdgeEval/constraint_applyer.rs` -> `src/EdgeEval/constraint_applier.rs`

## Up for discussion
Move `src/DataTypes/statepair_list.rs` into `src/ModelObjects/statepair_list.rs`
Move `src/Simulation/decision.rs` into `src/ModelObjects/decision.rs`
Move `src/Simulation/graph_layout` somewhere else. Maybe protobuf?

### PR DESCRIPTION
**Pull request by frederikja163**
closes #1 

Composes a bunch of stylistic changes to the repo.
Most of the edits consists of some file renamings and snake case renaming.

After merging this you might get a lot of warnings and merge conflicts on your own branches. However after merging this we can enforce a uniform naming scheme over all variables, fields and parameters in the entire repo.

PR Created by: @ECDAR-AAU-SW-P5/group6

### CODE REVIEW

**Comment by frederikja163** [comment]
I am aware of the build changes. Currently waiting on #7 in order to fix the pipeline after renaming the build directory.

**Review by Aavild — APPROVED** [review]

File: src/tests/save_component/composition_tests.rs [hunk 1]
```diff
@@ -6,117 +6,117 @@ mod test {
...
 
     #[test]
-    fn MachineSpecSavedRefinesSelf() {
+    fn machine_spec_saved_refines_self() {
         json_reconstructed_component_refines_base_self(ECDAR_UNI, "Machine || Spec");
     }
 
     #[test]
-    fn Machine2ResearcherSavedRefinesSelf() {
+    fn machine2researcher_saved_refines_self() {
```
Comment (Animal2662) [inline src/tests/save_component/composition_tests.rs:99]:
```suggestion
    fn machine_2_researcher_saved_refines_self() {
```

File: src/tests/save_component/composition_tests.rs [hunk 2]
```diff
@@ -6,117 +6,117 @@ mod test {
...
 
     #[test]
-    fn HalfAdm2Machine3SavedRefinesSelf() {
+    fn half_adm2machine3saved_refines_self() {
         json_reconstructed_component_refines_base_self(ECDAR_UNI, "HalfAdm2 || Machine3");
     }
 
     #[test]
-    fn HalfAdm2ResearcherSavedRefinesSelf() {
+    fn half_adm2researcher_saved_refines_self() {
```
Comment (Animal2662) [inline src/tests/save_component/composition_tests.rs:84]:
```suggestion
    fn half_adm_2_researcher_saved_refines_self() {
```

File: src/tests/save_component/composition_tests.rs [hunk 3]
```diff
@@ -6,117 +6,117 @@ mod test {
...
 
     #[test]
-    fn HalfAdm2Machine2SavedRefinesSelf() {
+    fn half_adm2machine2saved_refines_self() {
         json_reconstructed_component_refines_base_self(ECDAR_UNI, "HalfAdm2 || Machine2");
     }
 
     #[test]
-    fn HalfAdm2Machine3SavedRefinesSelf() {
+    fn half_adm2machine3saved_refines_self() {
```
Comment (Animal2662) [inline src/tests/save_component/composition_tests.rs:79]:
```suggestion
    fn half_adm_2_machine_3_saved_refines_self() {
```

File: src/tests/save_component/composition_tests.rs [hunk 4]
```diff
@@ -6,117 +6,117 @@ mod test {
...
 
     #[test]
-    fn HalfAdm2MachineSavedRefinesSelf() {
+    fn half_adm2machine_saved_refines_self() {
         json_reconstructed_component_refines_base_self(ECDAR_UNI, "HalfAdm2 || Machine");
     }
 
     #[test]
-    fn HalfAdm2Machine2SavedRefinesSelf() {
+    fn half_adm2machine2saved_refines_self() {
```
Comment (Animal2662) [inline src/tests/save_component/composition_tests.rs:74]:
```suggestion
    fn half_adm_2_machine_2_saved_refines_self() {
```

File: src/tests/save_component/composition_tests.rs [hunk 5]
```diff
@@ -6,117 +6,117 @@ mod test {
...
 
     #[test]
-    fn HalfAdm1ResearcherSavedRefinesSelf() {
+    fn half_adm1researcher_saved_refines_self() {
         json_reconstructed_component_refines_base_self(ECDAR_UNI, "HalfAdm1 || Researcher");
     }
 
     #[test]
-    fn HalfAdm2MachineSavedRefinesSelf() {
+    fn half_adm2machine_saved_refines_self() {
```
Comment (Animal2662) [inline src/tests/save_component/composition_tests.rs:69]:
```suggestion
    fn half_adm_2_machine_saved_refines_self() {
```

File: src/tests/save_component/composition_tests.rs [hunk 6]
```diff
@@ -6,117 +6,117 @@ mod test {
...
 
     #[test]
-    fn HalfAdm1Machine2SavedRefinesSelf() {
+    fn half_adm1machine2saved_refines_self() {
         json_reconstructed_component_refines_base_self(ECDAR_UNI, "HalfAdm1 || Machine2");
     }
 
     #[test]
-    fn HalfAdm1Machine3SavedRefinesSelf() {
+    fn half_adm1machine3saved_refines_self() {
```
Comment (Animal2662) [inline src/tests/save_component/composition_tests.rs:59]:
```suggestion
    fn half_adm_1_machine_3_saved_refines_self() {
```

**Review by sabotack — CHANGES_REQUESTED** [review]
Har lavet suggestions, men synes stadig der er enkelte filer, hvor der mangler en del underscores. 🫡

File: src/tests/save_component/no_operation_tests.rs [hunk 7]
```diff
@@ -6,96 +6,96 @@ mod test {
     const ECDAR_UNI: &str = "samples/json/EcdarUniversity";
 
     #[test]
-    fn T1SavedRefinesSelf() {
+    fn t1saved_refines_self() {
         json_reconstructed_component_refines_base_self(PATH, "Test1");
     }
     #[test]
-    fn T2SavedRefinesSelf() {
+    fn t2saved_refines_self() {
```
Comment (rasmusq) [inline src/tests/save_component/no_operation_tests.rs:13]:
```suggestion
    fn t2_saved_refines_self() {
```

File: src/tests/save_component/no_operation_tests.rs [hunk 8]
```diff
@@ -6,96 +6,96 @@ mod test {
...
         json_reconstructed_component_refines_base_self(PATH, "Test2");
     }
     #[test]
-    fn T3SavedRefinesSelf() {
+    fn t3saved_refines_self() {
```
Comment (rasmusq) [inline src/tests/save_component/no_operation_tests.rs:17]:
```suggestion
    fn t3_saved_refines_self() {
```

File: src/tests/save_component/no_operation_tests.rs [hunk 9]
```diff
@@ -6,96 +6,96 @@ mod test {
...
         json_reconstructed_component_refines_base_self(PATH, "Test3");
     }
     #[test]
-    fn T4SavedRefinesSelf() {
+    fn t4saved_refines_self() {
         json_reconstructed_component_refines_base_self(PATH, "Test4");
     }
     #[test]
-    fn T5SavedRefinesSelf() {
+    fn t5saved_refines_self() {
```
Comment (rasmusq) [inline src/tests/save_component/no_operation_tests.rs:25]:
```suggestion
    fn t5_saved_refines_self() {
```

File: src/tests/save_component/no_operation_tests.rs [hunk 10]
```diff
@@ -6,96 +6,96 @@ mod test {
...
         json_reconstructed_component_refines_base_self(PATH, "Test5");
     }
     #[test]
-    fn T6SavedRefinesSelf() {
+    fn t6saved_refines_self() {
```
Comment (rasmusq) [inline src/tests/save_component/no_operation_tests.rs:29]:
```suggestion
    fn t6_saved_refines_self() {
```

File: src/tests/save_component/no_operation_tests.rs [hunk 11]
```diff
@@ -6,96 +6,96 @@ mod test {
...
         json_reconstructed_component_refines_base_self(PATH, "Test6");
     }
     #[test]
-    fn T7SavedRefinesSelf() {
+    fn t7saved_refines_self() {
```
Comment (rasmusq) [inline src/tests/save_component/no_operation_tests.rs:33]:
```suggestion
    fn t7_saved_refines_self() {
```

File: src/tests/save_component/no_operation_tests.rs [hunk 12]
```diff
@@ -6,96 +6,96 @@ mod test {
...
         json_reconstructed_component_refines_base_self(PATH, "Test7");
     }
     #[test]
-    fn T8SavedRefinesSelf() {
+    fn t8saved_refines_self() {
```
Comment (rasmusq) [inline src/tests/save_component/no_operation_tests.rs:37]:
```suggestion
    fn t8_saved_refines_self() {
```

File: src/tests/save_component/no_operation_tests.rs [hunk 13]
```diff
@@ -6,96 +6,96 @@ mod test {
...
         json_reconstructed_component_refines_base_self(PATH, "Test8");
     }
     #[test]
-    fn T9SavedRefinesSelf() {
+    fn t9saved_refines_self() {
```
Comment (rasmusq) [inline src/tests/save_component/no_operation_tests.rs:41]:
```suggestion
    fn t9_saved_refines_self() {
```

File: src/tests/save_component/no_operation_tests.rs [hunk 14]
```diff
@@ -6,96 +6,96 @@ mod test {
...
         json_reconstructed_component_refines_base_self(PATH, "Test9");
     }
     #[test]
-    fn T10SavedRefinesSelf() {
+    fn t10saved_refines_self() {
```
Comment (rasmusq) [inline src/tests/save_component/no_operation_tests.rs:45]:
```suggestion
    fn t10_saved_refines_self() {
```

File: src/tests/save_component/no_operation_tests.rs [hunk 15]
```diff
@@ -6,96 +6,96 @@ mod test {
...
         json_reconstructed_component_refines_base_self(PATH, "Test10");
     }
     #[test]
-    fn T11SavedRefinesSelf() {
+    fn t11saved_refines_self() {
```
Comment (rasmusq) [inline src/tests/save_component/no_operation_tests.rs:49]:
```suggestion
    fn t11_saved_refines_self() {
```

File: src/tests/save_component/no_operation_tests.rs [hunk 16]
```diff
@@ -6,96 +6,96 @@ mod test {
...
         json_reconstructed_component_refines_base_self(PATH, "Test11");
     }
     #[test]
-    fn T12SavedRefinesSelf() {
+    fn t12saved_refines_self() {
```
Comment (rasmusq) [inline src/tests/save_component/no_operation_tests.rs:53]:
```suggestion
    fn t12_saved_refines_self() {
```

**Review by rasmusq — COMMENTED** [review]
Har lavet nogle suggestions.
Mange af dem er ekstra underscores.
Nogle er for at følge standarderne lidt bedre i cargo doc

**Review by rasmusq — APPROVED** [review]

**Review by sabotack — APPROVED** [review]

**Review by Aavild — APPROVED** [review]

**Review by williamwoldum — APPROVED** [review]

**Pull request merged**